src/praisonai-agents/praisonaiagents/tools/decorator.py

```
@@ -154,6 +156,31 @@ def __call__(self, *args, **kwargs) -> Any:
+            "function": {
+                "name": self.name,
+                "description": self.description,
+                "parameters": self.parameters
+            }
+        }
+        
+        # Apply dynamic override if present
+        if self._schema_override is not None:
+            try:
+                return self._schema_override(base_schema)
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)